File: tests/xcalablemp/NOT-IMPLEMENTED/23/array7.c
/*testp066.c*/
/*task指示文とarray指示文の組み合わせ*/ 
#include <xmp.h>
#include <stdio.h> 
#include <stdlib.h>     
static const int N=1000;
#pragma xmp nodes p(4)
#pragma xmp template t(0:N-1,0:N-1,0:N-1)
int m[4] = {200,100,400,300};
#pragma xmp distribute t(*,*, gblock(m)) onto p
int a[N],sa,ansa;
double b[N],sb,ansb;
char result;
int i;
#pragma xmp align a[i] with t(*,*,i)
#pragma xmp align b[i] with t(*,*,i)
#pragma xmp shadow a[1:0]
#pragma xmp shadow b[0:1]
int main(void){

   sa = 0;
   sb = 0.0;

#pragma xmp loop on t(*,*,i)
   for(i=0;i<N;i++){
      a[i]=1;
      b[i]=1.0;
   }

#pragma xmp task on p(1)
   {
#pragma xmp array on t(:,:,:)
      a[0:200] = 2;
#pragma xmp array on t(:,:,:)
      b[0:200] = 1.5;
   }
#pragma xmp loop on t(:,:,i)
   for(i=0;i<N;i++){
      sa = sa + a[i];
      sb = sb + b[i];
   }
#pragma xmp reduction(+:sa)
#pragma xmp reduction(+:sb)

   ansa = 1200;
   ansb = 1000.0+0.5*200;

   result = "OK";
   if(sa != ansa){
      result = "NG";
   }
   if(abs(sb-ansb) > 0.0000000001){
      result = "NG";
   }

   printf("%d %s %s\n",xmp_node_num(),"testp066.c",result);
   return 0;
}    
         
      
   

File: tests/XMP/NOT-IMPLEMENTED/23/array7.f90
! testp066.f
! task指示文とarray指示文の組み合わせ

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(4)
!$xmp template t(N,N,N)
!$xmp distribute t(*,*,gblock((/200,100,400,300/))) onto p
      integer a(N), sa, ansa
      real*8  b(N), sb, ansb
      character(len=2) result
!$xmp align a(i) with t(*,*,i)
!$xmp align b(i) with t(*,*,i)
!$xmp shadow a(1:0)
!$xmp shadow b(0:1)

      sa = 0
      sb = 0.0

!$xmp loop on t(:,:,i)
      do i=1, N
         a(i) = 1
         b(i) = 1.0
      enddo

!$xmp task on p(1)
!$xmp array on t(:,:,:)
      a(1:200) = 2
!$xmp array on t(:,:,:)
      b(1:200) = 1.5
!$xmp end task      
      
!$xmp loop on t(:,:,i)
      do i=1, N
         sa = sa+a(i)
         sb = sb+b(i)
      enddo

!$xmp reduction(+:sa)
!$xmp reduction(+:sb)

      ansa = N+200
      ansb = dble(N)+0.5*200

      result = 'OK'
      if(sa .ne. ansa) then
         result = 'NG'
      endif
      if(abs(sb-ansb) .gt. 0.00000000001) then
         result = 'NG'
      endif

      print *, xmp_node_num(), 'testp066.f ', result

      end
